tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -395,6 +451,200 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+			4.	Verify that pvc created in step 3 are bound
+			5.	Create a pod, say pod1 using pvc created in step 4.
+			6.	write some data to the volume.
+			7.	delete pod1.
+			8.	Relocate volumes from vmfs/vsand → vmfs/vsand.
+			9.	Recreate pod1
+			10.	Verify pod1 is running and pvc1 is accessible and verify the data written in step 6
+			11.	Delete pod1
+			12.	Delete pvc, sc and SPBM policy created for this test
+	*/
+	ginkgo.It("[csi-supervisor][csi-block-vanilla][csi-block-vanilla-parallelized][csi-guest]"+
+		" Relocate volume to another same type datastore", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		sharedvmfsURL, sharedvmfs2URL := "", ""
+		var datastoreUrls, allocationTypes []string
+		var policyName string
+
+		sharedvmfsURL = os.Getenv(envSharedVMFSDatastoreURL)
+		if sharedvmfsURL == "" {
+			ginkgo.Skip(fmt.Sprintf("Env %v is missing", envSharedVMFSDatastoreURL))
+		}
+
+		sharedvmfs2URL = os.Getenv(envSharedVMFSDatastore2URL)
+		if sharedvmfsURL == "" {
+			ginkgo.Skip(fmt.Sprintf("Env %v is missing", envSharedVMFSDatastore2URL))
+		}
+		datastoreUrls = append(datastoreUrls, sharedvmfsURL)
+		datastoreUrls = append(datastoreUrls, sharedvmfs2URL)
+
+		scParameters := make(map[string]string)
+		policyNames := []string{}
+		pvcs := []*v1.PersistentVolumeClaim{}
+		pvclaims2d := [][]*v1.PersistentVolumeClaim{}
+		scs := []*storagev1.StorageClass{}
+
+		rand.Seed(time.Now().UnixNano())
+		suffix := fmt.Sprintf("-%v-%v", time.Now().UnixNano(), rand.Intn(10000))
+		categoryName := "category" + suffix
+		tagName := "tag" + suffix
+
+		catID, tagID := createCategoryNTag(ctx, categoryName, tagName)
+		defer func() {
+			deleteCategoryNTag(ctx, catID, tagID)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfsURL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfsURL)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfs2URL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfs2URL)
+		}()
+
+		allocationTypes = []string{
+			lztAllocType,
+		}
+
+		ginkgo.By("create SPBM policy with thin/ezt volume allocation")
+		ginkgo.By("create a storage class with a SPBM policy created from step 1")
+		ginkgo.By("create a PVC each using the storage policy created from step 2")
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		var policyID *pbmtypes.PbmProfileId
+		for _, at := range allocationTypes {
+			policyID, policyName = createVmfsStoragePolicy(
+				ctx, pc, at, map[string]string{categoryName: tagName})
+
+			defer func() {
+				deleteStoragePolicy(ctx, pc, policyID)
+			}()
+
+			policyNames = append(policyNames, policyName)
+		}
+
+		if supervisorCluster {
+			assignPolicyToWcpNamespace(client, ctx, namespace, policyNames)
+		} else if guestCluster {
+			_, svNamespace := getSvcClientAndNamespace()
+			assignPolicyToWcpNamespace(client, ctx, svNamespace, policyNames)
+		}
+
+		for _, policyName := range policyNames {
```

Comment:
when there is only one allocationType we want to use in the test, better to remove loops at line 528, 546, 578 and 603